mlflow/keras.py

```diff
@@ -498,8 +501,11 @@ def fit_generator(self, *args, **kwargs):
         if early_stop_callback and early_stop_callback.restore_best_weights:
             stopped_epoch_metrics = {key: history.history[key][early_stop_callback.stopped_epoch]
                                      for key in history.history.keys()}
-            last_epoch = len(history.history['loss'])
-            try_mlflow_log(mlflow.log_metrics, stopped_epoch_metrics, step=last_epoch+1)
+            # Checking that a metric history exists
+            metric_key = next(iter(history.history))
```

Comment:
I think this actually throws instead of returning `None`:

```
(mlflow-python3) ➜  mlflow git:(python-mlflow-spark-autologging) python
Python 3.6.8 |Anaconda, Inc.| (default, Dec 29 2018, 19:04:46) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> empty_lst = []
>>> next(iter(empty_lst))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
StopIteration
```

Reply:
Well that's unfortunate. I'll add `None` as a default argument for `next` to fix it.